samples/ck-rx65n/basic-sample/src/iotconnect_app.c

```diff
@@ -0,0 +1,267 @@
+    for (int j = 0; j < i; j++) {
+        free(blocks[j]);
+    }
+}
+#endif /* MEMORY_TEST */
+
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
+static char* compose_device_id() {
+#define PREFIX_LEN (sizeof(DUID_PREFIX) - 1)
+    uint8_t mac_addr[6] = { 0 };
+    static char duid[PREFIX_LEN + sizeof(mac_addr) * 2 + 1 /* null */] = DUID_PREFIX;
+	for (int i = 0; i < sizeof(mac_addr); i++) {
+		sprintf(&duid[PREFIX_LEN + i * 2], "%02x", mac_addr[i]);
+	}
+    printf("DUID: %s\r\n", duid);
+    return duid;
+}
+#pragma GCC diagnostic pop
+
+static bool is_app_version_same_as_ota(const char *version) {
+    return strcmp(APP_VERSION, version) == 0;
+}
+
+static bool app_needs_ota_update(const char *version) {
+    return strcmp(APP_VERSION, version) < 0;
+}
+
+static void on_ota(IotclEventData data) {
+    const char *message = NULL;
+    bool needs_ota_commit = false;
+    char *url = iotcl_clone_download_url(data, 0);
+    bool success = false;
+    if (NULL != url) {
+        printf("Download URL is: %s\r\n", url);
+        const char *version = iotcl_clone_sw_version(data);
+        if (!version) {
+            printf("Failed to clone SW version! Out of memory?");
+            message = "Failed to clone SW version";
+        } else if (is_app_version_same_as_ota(version)) {
+            printf("OTA request for same version %s. Sending success\r\n", version);
+            success = true;
+            message = "Version is matching";
+        } else if (app_needs_ota_update(version)) {
+            printf("OTA update is required for version %s.\r\n", version);
+			success = false;
+			message = "Not implemented";
+        } else {
+            printf("Device firmware version %s is newer than OTA version %s. Sending failure\r\n", APP_VERSION,
+                    version);
+            // Not sure what to do here. The app version is better than OTA version.
+            // Probably a development version, so return failure?
+            // The user should decide here.
+            success = false;
+            message = "Device firmware version is newer";
+        }
+
+        free((void*) url);
+        free((void*) version);
+    } else {
+        // compatibility with older events
+        // This app does not support FOTA with older back ends, but the user can add the functionality
+        const char *command = iotcl_clone_command(data);
+        if (NULL != command) {
+            // URL will be inside the command
+            printf("Command is: %s\r\n", command);
+            message = "Old back end URLS are not supported by the app";
+            free((void*) command);
+        }
+    }
+    const char *ack = iotcl_create_ack_string_and_destroy_event(data, success, message);
+    if (NULL != ack) {
+        printf("Sent OTA ack: %s\r\n", ack);
+        iotconnect_sdk_send_packet(ack);
+        free((void*) ack);
+    }
+    if (needs_ota_commit) {

```

Comment:
This is leftover from copying OTA capable code. Please also fix in the cloudkit sample.

Reply:
Removed OTA related bits and return a not implemented message